pkgs/by-name/sk/skyscraper/package.nix

```
@@ -22,6 +28,17 @@ stdenv.mkDerivation (finalAttrs: {
     qt5.qmake
   ];
 
+  patchPhase = lib.optional xdg ''
+    runHook prePatch
+    substituteInPlace skyscraper.pro --replace "#DEFINES+=XDG" "DEFINES+=XDG"
+    runHook postPatch
+  '';
```

```suggestion
  postPatch = lib.optionalString xdg ''
    substituteInPlace skyscraper.pro --replace-fail "#DEFINES+=XDG" "DEFINES+=XDG"
  '';
```
Also the parameter would usually be called `enableXdg`.